Added setup instructions (#17)
edx/event-routing-backends: docs/getting_started.rst

Getting Started
===============

Setup
----------------

``event-routing-backends`` is developed as a pluggable application for the edx-platform. The code in this app
hooks into the `event-tracking`_ app that is installed as a part of
edx-platform. It provides new tracking backends and processors.

.. _event-tracking: https://github.com/edx/event-tracking

- Add ``event_routing_backends.apps.EventRoutingBackendsConfig`` in ``INSTALLED_APPS``.

> - Update ``EVENT_TRACKING_BACKENDS`` setting to include backend configuration. Sample configuration for ``caliper`` backend
> 
> Events can be filtered with `RegexFilter`_. For exampe in this configuration we allowing
> only enrollemnt, seek_video and edx.video.position.changed events to be routed to caliper event store.
> 
> .. _RegexFilter: https://github.com/edx/event-tracking/blob/master/eventtracking/processors/regex_filter.py

Reviewer: ```suggestion
- Update ``EVENT_TRACKING_BACKENDS`` setting to include backend configuration. See sample configuration for ``caliper`` backend below.

Events can be filtered with `RegexFilter`_. For example in this configuration we allowing
only enrollment, seek_video and edx.video.position.changed events to be routed to caliper event store.

.. _RegexFilter: https://github.com/edx/event-tracking/blob/master/eventtracking/processors/regex_filter.py
```

                    "caliper": {
                        "ENGINE": "event_routing_backends.backends.events_router.EventsRouter",
                        "OPTIONS": {
                            "processors": [
                                {
                                    "ENGINE": "event_routing_backends.processors.caliper.transformer_processor.CaliperProcessor",
                                    "OPTIONS": {}
                                },
                                {
                                    "ENGINE": "event_routing_backends.processors.caliper.envelop_processor.CaliperEnvelopProcessor",
                                    "OPTIONS": {
                                        "sensor_id": "http://test.com/sensors"
                                    }
                                }
                            ],
                            "backend_name": "caliper"
                        }
                    }
                }
            }
        }
    }

- Run migrations
   .. code-block:: bash

    $ ./manage.py lms migrate event_routing_backends

- Add router configuraton from django admin under ``EVENT_ROUTING_BACKENDS`` section

> Here is sample configuration for `Bearer Authentication`_ client which routes only those events where ``org_id`` is set to edX.

Reviewer: ```suggestion
Here is a sample configuration for a `Bearer Authentication`_ client which routes only those events where ``org_id`` is set to edX.
```

`override_args` allows us to pass any additional info in event.

.. _Bearer Authentication: https://swagger.io/docs/specification/authentication/bearer-authentication/

   .. code-block:: JSON

    [
       {
            "override_args": {
                "sensor": "test.sensor.com",
                "new_key": "new_value"
            },
            "host_configurations": {
                "auth_key": "test_key",
                "url": "http://concerned.host.com",
                "auth_scheme": "Bearer",
                "headers": {
                    "test": "header"
                }
            },
            "router_type": "AUTH_HEADERS",
            "match_params": {
                "context.org_id": "edX"
            }
        }
     ]

Local development
-----------------

If you have not already done so, create/activate a `virtualenv`_. Unless otherwise stated, assume all terminal code
below is executed within the virtualenv.

.. _virtualenv: https://virtualenvwrapper.readthedocs.org/en/latest/

Dependencies can be installed via the command below.

.. code-block:: bash

    $ make requirements

> Than you might want to run tests to make sure the setup went fine and there are no pre-existing problems (i.e. failed

Reviewer: ```suggestion
Then you might want to run tests to make sure the setup went fine and there are no pre-existing problems (i.e. failed
```
